src/colorizer/utils/data_utils.ts

```diff
@@ -5,6 +5,19 @@ import { FeatureThreshold, isThresholdCategorical, isThresholdNumeric, Threshold
 import ColorRamp from "../ColorRamp";
 import Dataset, { FeatureType } from "../Dataset";
 
+/** Returns whether the two arrays are deeply equal, where arr1[i] === arr2[i] for all i. */
+export function arrayDeepEquals<T>(arr1: T[], arr2: T[]): boolean {
```

Comment:
nit: tiny misnomer since this is only one layer deep and not all-the-way deep.  IDK if I have a better name in mind, maybe arrayElementsAreEqual, or maybe this name is ok...

Reply:
`arrayElementsAreEqual` makes sense to me. I'll make the change 👍